### PR TITLE
Records should not be rolled back in rollback() when the transaction state is marked as COMMITTED in 2PC

### PR DESCRIPTION
**Pull request by brfrn169**
Currently, in `TwoPhaseConsensusCommit.rollback()`, records are rolled back regardless of the transaction state, which could potentially lead to inconsistent data. This PR addresses this issue by checking the state and subsequently throwing `RollbackException` if the transaction state is COMMITTED. Please take a look!

### CODE REVIEW

File: core/src/test/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitTest.java [hunk 1]
```diff
@@ -335,4 +336,29 @@ public void rollback_CalledAfterCommitFails_ShouldNeverAbortStateAndRollbackReco
...
+  @Test
+  public void
+      rollback_UnknownTransactionStatusExceptionThrownByAbortState_ShouldThrowRollbackException()
+          throws UnknownTransactionStatusException, PreparationException {
+    // Arrange
+    transaction.prepare();
+    when(crud.getSnapshot()).thenReturn(snapshot);
+    when(commit.abortState(snapshot.getId())).thenThrow(UnknownTransactionStatusException.class);
+
+    // Act Assert
```
Comment (komamitsu) [inline core/src/test/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitTest.java:349]: Is it possible to verify `commit.rollbackRecords()` wasn't called?

Comment (brfrn169) [reply core/src/test/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitTest.java:349]: Add assertions for that in 806e19514b80c5b8de764a1a8841d482571a1fa3. Thanks.

**Review by komamitsu — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by brfrn169** [comment]
@Torch3333 I will merge this for now. Please take a look when you get a chance!

**Comment by brfrn169** [comment]
Pushed to the `3.9` branch as well.

**Review by Torch3333 — COMMENTED** [review]
LGTM, thank you!

**Pull request merged**